clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumerImpl.java

```diff
@@ -554,7 +555,7 @@ public void commitAsync() {
      */
     @Override
     public void setAcknowledgementCommitCallback(final AcknowledgementCommitCallback callback) {
-        throw new UnsupportedOperationException();
+        acknowledgementCommitCallbackHandler = new AcknowledgementCommitCallbackHandler(callback);
```

Comment:
If callback is null, I think you could just set `acknowledgementCommitCallbackHandler` to null.